File: golang/binaryTree-dfs/104.go
package main

// TreeNode Definition for a binary tree node.
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxDepth(root *TreeNode) int {
	if root == nil {
		return 0
	} else if root.Left == nil && root.Right == nil {
		return 1
	} else {
		return 1 + max(maxDepth(root.Left), maxDepth(root.Right))
	}
}

func max(a, b int) int {
	if a > b {
		return a
	}

	return b
}

File: python/binaryTree-dfs/104.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
        
def maxDepth(root: TreeNode) -> int:
    if not root:
        return 0
    elif not root.left and not root.right:
        return 1
    else:
        return 1 + max(maxDepth(root.left), maxDepth(root.right))
    

# Example: 
#       3
#     /   \
#    9     20
#         /  \
#        15   7    
root = TreeNode(3)
root.left = TreeNode(9)
root.right = TreeNode(20)
root.right.left = TreeNode(15)
root.right.right = TreeNode(7)
root.right.right.right = TreeNode(10)

# Call the maxDepth function and print the result
print(maxDepth(root))